cmake/Modules/LocalJpeg.cmake

```diff
@@ -2,22 +2,27 @@ set(AVIF_JPEG_TAG "3.0.4")
 
 add_library(JPEG::JPEG STATIC IMPORTED GLOBAL)
 
-set(LIB_DIR "${AVIF_SOURCE_DIR}/ext/libjpeg-turbo/build.libavif")
 if(MSVC)
-    set(LIB_FILENAME "${LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}jpeg-static${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(JPEG_STATIC_SUFFIX "-static")
 else()
-    set(LIB_FILENAME "${LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}jpeg${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(JPEG_STATIC_SUFFIX "")
 endif()
+set(LIB_BASENAME "${CMAKE_STATIC_LIBRARY_PREFIX}jpeg${JPEG_STATIC_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+set(LIB_DIR "${AVIF_SOURCE_DIR}/ext/libjpeg-turbo/build.libavif")
+set(LIB_FILENAME "${LIB_DIR}/${LIB_BASENAME}")
 if(EXISTS "${LIB_FILENAME}")
     message(STATUS "libavif(AVIF_JPEG=LOCAL): ${LIB_FILENAME} found, using for local JPEG")
     set(JPEG_INCLUDE_DIR "${AVIF_SOURCE_DIR}/ext/libjpeg-turbo")
 else()
     message(STATUS "libavif(AVIF_JPEG=LOCAL): ${LIB_FILENAME} not found, fetching")
     set(LIB_DIR "${CMAKE_CURRENT_BINARY_DIR}/libjpeg/src/libjpeg-build")
-    if(MSVC)
-        set(LIB_FILENAME "${LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}jpeg-static${CMAKE_STATIC_LIBRARY_SUFFIX}")
-    else()
-        set(LIB_FILENAME "${LIB_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}jpeg${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(LIB_FILENAME "${LIB_DIR}/${LIB_BASENAME}")
+    get_property(IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
```

Comment:
Vincent: While investigating this Visual Studio build error, I noticed that we are using `_deps\dav1d-install\lib\libdav1d.a` for all configuration types. This means we are installing the Debug, MinSizeRel, Release, RelWithDebInfo versions of libdav1d.a to the same location and use the installed libdav1d.a in all configuration types. This is not ideal.

Reply:
I agree. That can be fixed in a different CL. ExternalProject_Add is only used for dav1d (because it uses meson) and libjpeg-turbo (it officially states in does not support add_subdirectory, cf https://github.com/libjpeg-turbo/libjpeg-turbo/blob/main/BUILDING.md#sub-project-builds)